artiq/language/environment.py

```diff
@@ -337,21 +337,36 @@ def setattr_device(self, key):
         self.kernel_invariants = kernel_invariants | {key}
 
     @rpc(flags={"async"})
-    def set_dataset(self, key, value,
+    def set_dataset(self, key, value, *,
+                    unit=None, scale=None, precision=None,
                     broadcast=False, persist=False, archive=True):
         """Sets the contents and handling modes of a dataset.
 
         Datasets must be scalars (``bool``, ``int``, ``float`` or NumPy scalar)
         or NumPy arrays.
 
+        :param unit: A string representing the unit of the value.
+        :param scale: A numerical scaling factor by which the displayed value is
+            multiplied when referenced in the experiment.
```

Comment:
This should unambiguously describe what happens here, i.e. that it is inferred for the given unit by default, and only needs to be specified if the unit is not known, or a custom scaling is desired. It might be worth linking to the `NumberValue` documentation here as well (or the other way round).

Reply:
As far as I remember it is copy-pasted from the NumberValue docstring. Some updated docstrings on both may be worthwhile.